utils/json-utils/src/test/scala/pl/touk/nussknacker/engine/json/encode/BestEffortJsonSchemaEncoderTest.scala

```diff
@@ -121,8 +122,40 @@ class BestEffortJsonSchemaEncoderTest extends AnyFunSuite {
   }
 
   test("should encode number") {
-    val encoded = encoder.encodeWithJsonValidation(1L, schemaNumber)
-    encoded shouldEqual Valid(Json.fromLong(1L))
+    forAll(Table(
+      ("data", "schema", "expected"),
+      (1, schemaNumber, Json.fromLong(1L)),
+      (1, schemaIntegerNumber, Json.fromLong(1L)),
+      (1.0d, schemaNumber, Json.fromDoubleOrNull(1.0)),
+      (1.0d, schemaIntegerNumber, Json.fromLong(1L)),
+      (1.0f, schemaNumber, Json.fromFloatOrNull(1.0f)),
+      (1.0f, schemaIntegerNumber, Json.fromLong(1L)),
+      (BigDecimal.valueOf(1.0), schemaNumber, Json.fromBigDecimal(BigDecimal.valueOf(1.0))),
+      (java.math.BigDecimal.valueOf(1.0), schemaNumber, Json.fromBigDecimal(BigDecimal.valueOf(1.0))),
+      (BigDecimal.valueOf(1.0), schemaIntegerNumber, Json.fromLong(1L)),
+      (java.math.BigDecimal.valueOf(1.0), schemaIntegerNumber, Json.fromLong(1L)),
+      (BigInt.long2bigInt(1), schemaNumber, Json.fromBigInt(BigInt.long2bigInt(1))),
+      (java.math.BigInteger.valueOf(1), schemaNumber, Json.fromBigInt(BigInt.long2bigInt(1))),
+      (BigInt.long2bigInt(1), schemaIntegerNumber, Json.fromLong(1L)),
+      (java.math.BigInteger.valueOf(1), schemaIntegerNumber, Json.fromLong(1L)),
+    )) { (data, schema, expected) =>
+      encoder.encodeWithJsonValidation(data, schema) shouldBe Valid(expected)
+    }
+  }
+
+  test("should throw proper error trying encode integer number") {
+    forAll(Table(
+      ("data", "schema", "expected"),
+      (1.6d, schemaIntegerNumber, invalid(s"Not expected type: java.lang.Double for field with schema: $schemaIntegerNumber.")),
```

Comment:
I think this message is a bit misleading, as the problem is not with data type, but with fraction/number of digits, e.g.   `1.0d` would work...

Reply:
Above is test with 0 as well:
`
      (1.0d, schemaNumber, Json.fromDoubleOrNull(1.0)),
      (1.0d, schemaIntegerNumber, Json.fromLong(1L)),
      (1.0f, schemaNumber, Json.fromFloatOrNull(1.0f)),
      (1.0f, schemaIntegerNumber, Json.fromLong(1L)),
`